src/routes/i18nRedirect/containers/I18nRedirect.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import React from 'react'
+import PropTypes from 'prop-types'
+import { redirect } from 'redux-first-router'
+import { connect } from 'react-redux'
+import CityModel from '../../../modules/endpoint/models/CityModel'
+import { goToLanding } from '../../../modules/app/routes/landing'
+import { goToCategories } from '../../../modules/app/routes/categories'
+import { MAX_LANGUAGE_CODE_LENGTH, MIN_LANGUAGE_CODE_LENGTH } from '../../../modules/app/constants'
+import { goToNotFound } from '../../../modules/app/routes/notFound'
+
+import type { Action } from 'redux-first-router/dist/flow-types'
+import { goToMainDisclaimer } from '../../../modules/app/routes/mainDisclaimer'
+
+type Props = {
+  redirect: (Action) => void,
+  cities: Array<CityModel>,
+  param: ?string
+}
+
+/**
+ * Adds the language code at the end of the current path
+ */
+export class I18nRedirect extends React.Component<Props> {
+  static contextTypes = {
+    // we have to do this with PropTypes, because context is not known at compile time. A possible solution would be:
+    // https://github.com/codemix/flow-runtime
+    i18n: PropTypes.object.isRequired
+  }
+
+  getRedirectAction (): Action {
+    const {param, cities} = this.props
+    const i18n = this.context.i18n
+
+    // the param does not exist (or is 'landing'), so redirect to the landing page with the detected language
+    if (!param || param === 'landing') {
+      return goToLanding(i18n.language)
+    }
+
+    if (param === 'disclaimer') {
+      return goToMainDisclaimer()
+    }
+
+    if (param === 'not_found') {
+      return goToNotFound()
+    }
```

Comment:
⚠️ We shouldn't need to care about disclaimer or not_found here. I18nRedirect should only be rendered, when it's needed.

I think the main issue is, that the router still maps '/disclaimer' to the i18nRoute, and not to the mainDisclaimerRoute... This should not happen. I hope we can configure this in the router itself.

Reply:
yep you're right I also stumbled about this right now. As it was with ReduxLittleRouter, the order of the routes is important (in the RoutesMap). So if you put the disclaimer route before i18nRedirect, the disclaimer route is used when entering disclaimer

Reply:
should work again now, fixed the issue in WEBAPP-223